classes/mail/Mailable.php

```diff
@@ -451,23 +455,43 @@ public static function getDataDescriptions(): array
             }
         }
 
-        foreach ($args as $arg) { /** @var ReflectionParameter $arg) */
-            $argClassName = $arg->getType()->getName();
-
+        foreach ($args as $arg) {
+            $argTypes = static::getTypeNames($arg->getType());
             foreach ($map as $dataClass => $variableClass) {
-                if (is_a($argClassName, $dataClass, true)) {
-                    $descriptions = array_merge(
-                        $descriptions,
-                        $variableClass::descriptions()
-                    );
-                    continue 2;
+                foreach ($argTypes as $argType) {
+                    if (is_a($argType, $dataClass, true)) {
+                        $descriptions = array_merge(
+                            $descriptions,
+                            $variableClass::descriptions()
+                        );
+                        // An intersection type could be passed in (e.g. Submission & PublishedSubmission), so allow the code check other maps for correctness
+                        continue 2;
+                    }
                 }
             }
         }
 
         return $descriptions;
     }
 
+    /**
+     * Retrieve the list of type names that might compose a given type
+     * @return string[]
+     */
+    protected static function getTypeNames(ReflectionType $type): array
```

Comment:
This is supposed to retrieve all types that compose the argument, e.g.:
```php
Mailable::getTypeNames((new ReflectionFunction(fn (int|bool|(stdClass&Application) $a) => 0))->getParameters()[0]->getType());
```
Retrieves ['stdClass', 'APP\core\Application', 'int', 'bool']